Add peggy.code-workspace

markw65 opened the pull request:
See discussion on #42 

It turns out that you can include recommended extensions directly in the .code-workspace file, so I didn't need to add a .vscode/extensions.json file.

To test this:

 - I downloaded the insiders build of vscode. This has its own global settings, and list of installed extensions, so I was effectively starting from a fresh vscode install
 - I cloned a fresh copy of `@markw65/peggy`, and checked out `origin/peggy-code-workspace`.
 - I ran `npm install`, then opened the peggy folder from vscode-insiders.
 - It popped up a non-blocking notice that the folder contains a code-workspace file, and offered to open it, which I did.
 - It popped up a non-blocking notice asking if I wanted to install the recommended extensions, which I did (there was also an option to just show the recommended extensions).
 - I opened `generate-js.js`, and made some changes that eslint didn't like, and hit Shift-Option-F to format, and it fixed them for me, without modifying the rest of the file.
 - To confirm that the code-workspace settings can be overridden, I went to settings, and set `defaulltFormatter` to `Typescript and Javascript Language Features` in the `peggy` folder. This created a .vscode/settings.json file (which was hidden from git by the new .gitignore). I formatted `generate-js.js` again, and it changed lots of things in ways that lint didn't like (so that formatter isn't actually useful, but demonstrates that people can still choose to use something other than the default).

Given that settings can be overridden via .vscode/settings.json, I'm inclined to turn on `Format on Save` in `peggy.code-workspace` too.

markw65 commented on peggy.code-workspace:
```diff
@@ -0,0 +1,18 @@
+	],
+	"settings": {
+		"eslint.format.enable": true,
+		"editor.formatOnSave": true,
```
I didn't mean to leave this in...

hildjj commented:
peggy.code-workspace probably needs to be added to .npmignore.  Check with `npm pack --dry-run`

hildjj commented:
Note: I'll likely close #42 after this lands, after playing with it a little.

markw65 commented:
I did need to add peggy.code-workspace to .npmignore, and while there I realized that I added tsconfig*.json, to cover the tsconfig-base.json that I added in #435.

hildjj commented:
needs a changelog entry still, please.
